src/firefly/java/edu/caltech/ipac/firefly/server/visualize/imageretrieve/URLFileRetriever.java

```diff
@@ -44,6 +44,9 @@
     public FileInfo getFile(WebPlotRequest request) throws FailedRequestException, GeomException, SecurityException {
         FileInfo fitsFileInfo;
         String urlStr = request.getURL();
+        if (urlStr.toLowerCase().startsWith("file:///")) {
+            return new LocalFileRetriever().getFileByName(urlStr.substring(7));
+        }
         if (urlStr == null) throw new FailedRequestException("Could not find file", "request.getURL() returned null");
```

Comment:
should the line "if (urlStr == null) ... " be checked before the line "if (urlStr.toLowerCase().startsWith()....." ?